File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/inputrepresentation/RepresentationFactory.java
package org.akoshterek.backgammon.agent.inputrepresentation;

import org.akoshterek.backgammon.agent.raw.RawRepresentation;

/**
 * @author Alex
 *         date 22.09.2015.
 */
public class RepresentationFactory {
    public static InputRepresentation createInputRepresentation(String fullName) {
        String fullNameLower = fullName.toLowerCase();
        String[] tokens = fullNameLower.split("-");
        String representationName = tokens[0];
        String codecName = tokens[1];
        PointCodec codec = createCodec(codecName);
        switch (representationName) {
            case "raw":
                return new RawRepresentation(codec);
            default:
                throw new IllegalArgumentException("Unknown input representation " + representationName);
        }
    }

    private static PointCodec createCodec(String codecName) {
        switch (codecName) {
            case "sutton":
                return new SuttonCodec();
            case "tesauro89":
                return new Tesauro89Codec();
            case "tesauro92":
                return new Tesauro92Codec();
            case "gnubg":
                return new GnuBgCodec();
            default:
                throw new IllegalArgumentException("Unknown point codec " + codecName);
        }
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/inputrepresentation/RepresentationFactory.scala
package org.akoshterek.backgammon.agent.inputrepresentation

import org.akoshterek.backgammon.agent.raw.RawRepresentation

object RepresentationFactory {
  def createInputRepresentation(fullName: String): InputRepresentation = {
    val fullNameLower: String = fullName.toLowerCase
    val tokens: Array[String] = fullNameLower.split("-")
    val representationName: String = tokens(0)
    val codecName: String = tokens(1)
    val codec: PointCodec = createCodec(codecName)
    representationName match {
      case "raw" => new RawRepresentation(codec)
      case _ => throw new IllegalArgumentException("Unknown input representation " + representationName)
    }
  }

  private def createCodec(codecName: String): PointCodec = {
    codecName match {
      case "sutton" => SuttonCodec
      case "tesauro89" => Tesauro89Codec
      case "tesauro92" => Tesauro92Codec
      case "gnubg" => GnuBgCodec
      case _ => throw new IllegalArgumentException("Unknown point codec " + codecName)
    }
  }
}